frontend/packages/dev-console/src/components/custom-resource-list/CustomResourceList.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Table, TextFilter } from '@console/internal/components/factory';
+import { SortByDirection } from '@patternfly/react-table';
+import { CheckBoxes } from '@console/internal/components/row-filter';
+import { getQueryArgument } from '@console/internal/components/utils';
+import {
+  CustomResourceListFilterType,
+  CustomResourceListProps,
+} from './custom-resource-list-types';
+
+const CustomResourceList: React.FC<CustomResourceListProps> = ({
+  dependentResource,
+  items,
+  queryArg,
+  rowFilters,
+  getFilteredItems,
+  resourceHeader,
+  resourceRow,
+}) => {
+  const [listItems, setListItems] = React.useState([]);
+  const [filteredListItems, setFilteredListItems] = React.useState([]);
+  const [fetched, setFetched] = React.useState(false);
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const queryArgument = getQueryArgument(queryArg);
+    const activeFilters = queryArgument?.split(',');
+
+    const fetchListItems = async () => {
+      let newListItems: any;
+      try {
+        newListItems = await items;
```

Comment:
I would rather rename this to be `fetchCustomResources` and make it a function that returns a promise to make it more clear that this is used to fetch the resources.